File: rider-extension/src/main/kotlin/com/yeahbah/efvibe/services/PathResolver.kt
package com.yeahbah.efvibe.services

import com.intellij.openapi.project.Project
import java.nio.file.Path
import kotlin.io.path.absolutePathString
import kotlin.io.path.isAbsolute
import kotlin.io.path.normalize

object PathResolver {
    fun solutionDirectory(project: Project): Path =
        Path.of(project.basePath ?: System.getProperty("user.dir"))

    fun resolve(value: String, project: Project): String {
        val trimmed = value.trim()
        if (trimmed.isEmpty()) return ""

        val base = solutionDirectory(project)
        val expanded = trimmed
            .replace("\${workspaceFolder}", base.absolutePathString())
            .replace("\$(SolutionDir)", ensureTrailingSeparator(base.absolutePathString()))

        val path = Path.of(expanded)
        return if (path.isAbsolute()) {
            path.normalize().absolutePathString()
        } else {
            base.resolve(path).normalize().absolutePathString()
        }
    }

    private fun ensureTrailingSeparator(path: String): String =
        if (path.endsWith('/') || path.endsWith('\\')) path else "$path/"
}

File: visualstudio-extension/Services/PathResolver.cs
using System;
using System.IO;

namespace MyEfVibe.VisualStudio.Services;

internal static class PathResolver
{
    internal static string ResolvePath(string value, string solutionDirectory)
    {
        var trimmed = (value ?? string.Empty).Trim();

        if (trimmed.Length == 0)
            return string.Empty;

        var expanded = trimmed
            .Replace("$(SolutionDir)", EnsureTrailingSeparator(solutionDirectory))
            .Replace("$(solutionDir)", EnsureTrailingSeparator(solutionDirectory))
            .Replace("${workspaceFolder}", solutionDirectory);

        expanded = Environment.ExpandEnvironmentVariables(expanded);

        if (Path.IsPathRooted(expanded) || string.IsNullOrWhiteSpace(solutionDirectory))
            return Path.GetFullPath(expanded);

        return Path.GetFullPath(Path.Combine(solutionDirectory, expanded));
    }

    private static string EnsureTrailingSeparator(string directory)
    {
        if (string.IsNullOrWhiteSpace(directory))
            return string.Empty;

        return directory.EndsWith(Path.DirectorySeparatorChar.ToString(), StringComparison.Ordinal)
            ? directory
            : directory + Path.DirectorySeparatorChar;
    }
}
